From c5d9c8be2bda087ce33b36f46971dfb77f976827 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonas=20=C3=85dahl?= <jadahl@gmail.com>
Date: Mon, 6 Apr 2020 14:20:43 +0200
Subject: [PATCH] wayland: Always store unconfigured size

This means it'll always be as up to date GdkWindow::width/height. We
still skip the resize for non-configured windows though, to avoid
mapping with the wrong size.

The commit f06ee688fe86ec8e1db1c515a7bf2ccc8decc310 also accidentally
removed the unconfigured size setting completely, so this essentially
adds it back, but always sets it.

Closes: https://gitlab.gnome.org/GNOME/gtk/-/issues/2582
---
 gdk/wayland/gdkwindow-wayland.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gdk/wayland/gdkwindow-wayland.c b/gdk/wayland/gdkwindow-wayland.c
index f6acc74366..30d4353826 100644
--- a/gdk/wayland/gdkwindow-wayland.c
+++ b/gdk/wayland/gdkwindow-wayland.c
@@ -1156,6 +1156,8 @@ gdk_wayland_window_maybe_configure (GdkWindow *window,
   gboolean is_xdg_popup;
   gboolean is_visible;
 
+  impl->unconfigured_width = calculate_width_without_margin (window, width);
+  impl->unconfigured_height = calculate_height_without_margin (window, height);
 
   if (should_inhibit_resize (window))
     return;
-- 
2.24.1

